test/lambda-grep.py

```diff
@@ -0,0 +1,85 @@
+import json
+import queue
+import string
+import threading
+import time
+
+import boto3
+
+lambda_client = boto3.client('lambda')
+
+
+def lambda_grep(file_path, phrase):
+    """Call and return the result of an AWS Lambda call of grep on file_path.
+
+    Returns:
+        a dictionary of the Lambda response
+        on success this will  include "matches",
+        on error this will include "error_msg"
+    """
+    test_event = {
+        "file_path": file_path,
+        "phrase": phrase
+    }
+
+    invoke_response = lambda_client.invoke(
+        FunctionName="SQLiteDemo-ssqlite-test",
+        InvocationType='RequestResponse',
+        Payload=json.dumps(test_event)
+    )
+    return json.loads(invoke_response['Payload'].read().decode("utf-8"))
+
+
+def process_queue(q):
+    """Continually print the result of Lambda grep calls."""
+    while not q.empty():
+        file_path, phrase = q.get()
+        lambda_result = lambda_grep(file_path, phrase)
+        lines = lambda_result.get('matches')
+        if lines:
+            print(''.join(lines))
+        else:
+            err_type = lambda_result.get('errorType')
+            err_msg = lambda_result.get('errorMessage')
+            if err_type and err_msg:
+                print(f"{file_path}: {err_type} - {err_msg}\n")
+        q.task_done()
+
+
+def lambda_grep_multiple(file_paths, phrase, lambda_concurrency=4):
```

Comment:
Not quite sure the best way to concurrently call a bunch of lambda functions - I used multiple threads here that pull from a queue.

Reply:
Have you verified that they are all working in parallel? Threads are a reasonable way but we always have to worry about the Global Interpreter Lock (GIL) in Python. So you could also think about using `multiprocessing` instead.